tests/generation/test_utils.py

```diff
@@ -1739,6 +1739,88 @@ def test_prompt_lookup_decoding_matches_greedy_search(self):
             for output in (output_greedy, output_prompt_lookup):
                 self._check_outputs(output, input_ids, model.config, use_cache=True)
 
+    def test_dola_decoding_sample(self):
+        # This test ensures that the DoLa generation does not introduce output changes over greedy search.
+        # NOTE (1): The sentence above is true most of the time, there is a tiny difference in the logits due to matmul
+        # shape differences -- and it may result in a different output. The input shape difference happens in the
+        # main model, that runs the forward pass with several candidates at once (as opposed to generating one token at
+        # a time). See https://github.com/huggingface/transformers/issues/25420#issuecomment-1775317535 for more info.
+        # NOTE (2): It breaks the pattern in the tests above, for multiple reasons:
+        # - dola_decoding does not support `batch_size > 1`
+        # - dola_decoding does not support encoder-decoder models
+
+        for model_class in self.all_generative_model_classes:
+            if any(model_name in model_class.__name__.lower() for model_name in ["fsmt", "reformer"]):
+                self.skipTest("Won't fix: old model with different cache format")
```

Comment:
Fixed! `fsmt` can pass the test, while `reformer` has a new issue of mismatched early layer hidden size vs lm_head input size. Thus I created another category of skip to avoid reformer.